pkg/util/etcdutil/migrate.go

```diff
@@ -0,0 +1,87 @@
+package etcdutil
+
+import (
+	"bytes"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/golang/glog"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/client/restclient"
+	"k8s.io/kubernetes/pkg/client/unversioned"
+	"k8s.io/kubernetes/pkg/labels"
+)
+
+func Migrate() error {
+	time.Sleep(30 * time.Second)
+	// TODO: poll if TPR ready?
+	fmt.Println("etcd TPR is ready ===")
+
+	kubecli, err := unversioned.New(&restclient.Config{
+		Host: "http://127.0.0.1:8080",
+	})
+	if err != nil {
+		return err
+	}
+
+	ip := ""
+	for {
+		podList, err := kubecli.Pods("kube-system").List(api.ListOptions{
+			LabelSelector: labels.SelectorFromSet(labels.Set{"k8s-app": "boot-etcd"}),
+		})
+		if err != nil {
+			glog.Errorf("fail to list 'boot-kube' pod, retrying: %v", err)
+			time.Sleep(5 * time.Second)
```

Comment:
i feel we should make the list a func. if the func return error we will do sleep. now we have 3 sleeps in this for loop, which seems to be crazy.